File: ProjectEuler/p2.py
# Even Fibonacci numbers
def main():
    i1, i2 = 1, 2
    sum = 0
    while i2 <= 4000000:
        if i2 % 2 == 0:
            sum += i2
        i1, i2 = i2, i1 + i2
    return sum

if __name__ == "__main__":
    print(main())

File: Project Euler/p2.cpp
// Even Fibonacci numbers
#include <bits/stdc++.h>

using namespace std;

int main() {
    int sum = 0;
    int i1 = 1, i2 = 2;

    while (i2 <= 4000000) {
        if (i2 % 2 == 0) {
            sum += i2;
        }
        int temp = i1;
        i1 = i2;
        i2 = temp + i2;
    }

    return sum;
}
